tool/tsh/tsh.go

```diff
@@ -3418,7 +3424,7 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 	// Handle gracefully if the profile is empty, the key cannot
 	// be found, or the key isn't supported as an agent key.
 	if profileSiteName != "" {
-		if err := tc.LoadKeyForCluster(profileSiteName); err != nil {
+		if err := tc.LoadKeyForCluster(cf.Context, profileSiteName); err != nil {
```

Comment:
nit: Should we use the tracer ctx here?
```suggestion
		if err := tc.LoadKeyForCluster(ctx, profileSiteName); err != nil {
```